catalog/registry/catalog-registry-metacard-filter/src/main/java/ddf/catalog/registry/metacard/filter/RegistryQueryDelegate.java

```diff
@@ -0,0 +1,475 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.catalog.registry.metacard.filter;
+
+import java.util.Date;
+import java.util.List;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.filter.FilterDelegate;
+
+public class RegistryQueryDelegate extends FilterDelegate<Boolean> {
+
+    /*
+    Returns false if this is a query that should not return registry metacards
+    Return true if this is a query that should return registry metacards
+     */
+
+    @Override
+    public Boolean and(List<Boolean> operands) {
+        return operands.stream()
+                .reduce(false, (a, b) -> a || b);
```

Comment:
`and` is a confusing case, because a query for registry metacards (true) ANDed with a query that isn't explicitly for registry metacards (false) should return true (will return registry metacards), right?